File: androidApp/src/main/kotlin/subha/app/cyra/ui/components/CyraRadioOptionRow.kt
package subha.app.cyra.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.interaction.MutableInteractionSource
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.remember
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.semantics.Role
import androidx.compose.ui.unit.dp
import subha.app.cyra.ui.theme.CyraTheme

/**
 * A full-row single-select option - leading icon, label, trailing radio dot - used for
 * marital status. Same 16dp corner radius as [CyraTextField] for visual consistency
 * across the flow's controls, not the smaller radius a first draft used.
 */
@Composable
fun CyraRadioOptionRow(
    icon: @Composable () -> Unit,
    label: String,
    selected: Boolean,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
) {
    val borderColor = if (selected) MaterialTheme.colorScheme.primary else MaterialTheme.colorScheme.outline
    val backgroundColor = if (selected) MaterialTheme.colorScheme.primaryContainer.copy(alpha = 0.35f) else MaterialTheme.colorScheme.surface

    Row(
        modifier = modifier
            .fillMaxWidth()
            .background(color = backgroundColor, shape = RoundedCornerShape(16.dp))
            .border(width = 1.dp, color = borderColor, shape = RoundedCornerShape(16.dp))
            .clickable(
                interactionSource = remember { MutableInteractionSource() },
                indication = null,
                role = Role.RadioButton,
                onClick = onClick,
            )
            .padding(horizontal = 16.dp, vertical = 14.dp),
        horizontalArrangement = Arrangement.SpaceBetween,
        verticalAlignment = Alignment.CenterVertically,
    ) {
        Row(horizontalArrangement = Arrangement.spacedBy(12.dp), verticalAlignment = Alignment.CenterVertically) {
            icon()
            Text(text = label, style = MaterialTheme.typography.titleMedium, color = MaterialTheme.colorScheme.onSurface)
        }
        RadioDot(selected = selected)
    }
}

@Composable
private fun RadioDot(selected: Boolean) {
    val borderColor = if (selected) MaterialTheme.colorScheme.primary else MaterialTheme.colorScheme.outline
    Box(
        modifier = Modifier
            .size(20.dp)
            .border(width = 1.6.dp, color = borderColor, shape = CircleShape),
        contentAlignment = Alignment.Center,
    ) {
        if (selected) {
            Box(
                modifier = Modifier
                    .size(10.dp)
                    .background(color = MaterialTheme.colorScheme.primary, shape = CircleShape),
            )
        }
    }
}

@CyraPreviews
@Composable
private fun CyraRadioOptionRowPreview() {
    CyraTheme {
        Column(modifier = Modifier.padding(16.dp), verticalArrangement = Arrangement.spacedBy(10.dp)) {
            CyraRadioOptionRow(icon = { HeartIcon() }, label = "Single", selected = false, onClick = {})
            CyraRadioOptionRow(icon = { LockIcon() }, label = "Prefer not to say", selected = true, onClick = {})
        }
    }
}

File: iosApp/iosApp/Components/CyraRadioOptionRow.swift
import SwiftUI

/// A full-row single-select option - leading SF Symbol, label, trailing radio dot -
/// mirrors Android's `CyraRadioOptionRow.kt`. Used for marital status. Same 16pt corner
/// radius as `CyraTextField` for visual consistency across the flow's controls.
struct CyraRadioOptionRow: View {
    let systemImage: String
    let label: String
    let selected: Bool
    let onClick: () -> Void

    var body: some View {
        Button(action: onClick) {
            HStack {
                HStack(spacing: 12) {
                    Image(systemName: systemImage)
                        .foregroundColor(.cyraOnSurfaceVariant)
                        .frame(width: 20)
                    Text(label)
                        .font(CyraFont.titleMedium())
                        .foregroundColor(.cyraOnSurface)
                }
                Spacer()
                ZStack {
                    Circle()
                        .stroke(selected ? Color.cyraPrimary : Color.cyraOutline, lineWidth: 1.6)
                        .frame(width: 20, height: 20)
                    if selected {
                        Circle()
                            .fill(Color.cyraPrimary)
                            .frame(width: 10, height: 10)
                    }
                }
            }
            .padding(.horizontal, 16)
            .padding(.vertical, 14)
            .background(
                RoundedRectangle(cornerRadius: 16)
                    .fill(selected ? Color.cyraPrimaryContainer.opacity(0.35) : Color.cyraSurface),
            )
            .overlay(
                RoundedRectangle(cornerRadius: 16)
                    .stroke(selected ? Color.cyraPrimary : Color.cyraOutline, lineWidth: 1),
            )
        }
        .buttonStyle(.plain)
    }
}

#Preview {
    VStack(spacing: 10) {
        CyraRadioOptionRow(systemImage: "heart", label: "Single", selected: false, onClick: {})
        CyraRadioOptionRow(systemImage: "lock", label: "Prefer not to say", selected: true, onClick: {})
    }
    .padding(24)
    .cyraThemed()
}
